### PR TITLE
[Bugfix] handle sort field as field

### LINKED ISSUE #40332: Sort Feature in Atlas and Report fails if attribute column name has space
**Describe the bug**
This doesn't appear to have been reported before as far as I can tell, but if you try to use the sort feature in either the Atlas or Report, if there is a space in the attribute column label, then the sort fails and it defaults to sorting by fid. This occurs in a geopackage file btw.

**How to Reproduce**
So 'sort_order' as an attribute table column label works but 'sort order' fails

**QGIS and OS versions**
QGIS 3.16.x and 3.10.x
macOS Catalina and Mojave
Windows 10

Thanks again to the team for all the effort in QGIS. 

Luke

### PR DESCRIPTION
**Pull request by roya0045**
## Description

Proposal bugfix. to ensure column name are treated as fields and not as a raw expression.

Fixes #40332

### CODE REVIEW

**Comment by nyalldawson** [comment]
Did you build and test this locally?

(spoiler: you didn't)

**Comment by roya0045** [comment]
> Did you build and test this locally?
> 
> (spoiler: you didn't)

Still working on fixing my local build, that's why I made this as a draft to not forget about it over the holidays. ;)

File: src/gui/layout/qgslayoutatlaswidget.cpp [hunk 1]
```diff
@@ -240,7 +240,11 @@ void QgsLayoutAtlasWidget::changesSortFeatureExpression( const QString &expressi
 
   mBlockUpdates = true;
   mLayout->undoStack()->beginCommand( mAtlas, tr( "Change Atlas Sort" ) );
-  mAtlas->setSortExpression( expression );
+  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( mAtlasCoverageLayerComboBox->currentLayer() );
+  if ( QgsExpression::expressionToLayerFieldIndex( expression, vlayer ) != -1 && !expression.contains( '\"' ) )
```
Comment (nyalldawson) [inline src/gui/layout/qgslayoutatlaswidget.cpp:244]: @roya0045 given that we're likely to need this fix in many different places (eg https://github.com/qgis/QGIS/pull/46546 ), could we extract this logic out to a `static QString QgsExpression::quoteExpressionIfNeeded( const QString& expression, QgsVectorLayer* layer = nullptr )` method?

Comment (roya0045) [reply src/gui/layout/qgslayoutatlaswidget.cpp:244]: I have no issues doing it, should be done in the coming days.

Comment (roya0045) [reply src/gui/layout/qgslayoutatlaswidget.cpp]: Implemented, did the tests too, though I was unsure if leaving the whitespaces at the start and end is a good choice. Are whitespaces in fields critical? I would propose calling 'trimmed' on the string, either in the columnrefquote of before passing the string to it. 

What's your stance on trimming the strings?

**Comment by roya0045** [comment]
fine

**Comment by nyalldawson** [comment]
Superseded by https://github.com/qgis/QGIS/pull/47083

**Pull request closed without merging**